test/functional/p2p_getaddr_caching.py

```diff
@@ -44,13 +43,15 @@ def addr_received(self):
 class AddrTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
-        # Start onion ports after p2p and rpc ports.
-        port = PORT_MIN + 2 * PORT_RANGE
-        self.onion_port1 = port
-        self.onion_port2 = port + 1
+
+    def setup_network(self):
```

Comment:
any need to overwrite this method? `PortSeed.n` should be possible to set right after parsing the options:


```diff
diff --git a/test/functional/test_framework/test_framework.py b/test/functional/test_framework/test_framework.py
index a39ee003ef..3f02d21d42 100755
--- a/test/functional/test_framework/test_framework.py
+++ b/test/functional/test_framework/test_framework.py
@@ -223,11 +223,11 @@ class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
                 # It still needs to exist and be None in order for tests to work however.
                 self.options.descriptors = None
 
+        PortSeed.n = self.options.port_seed
+
     def setup(self):
         """Call this method to start up the test framework object with options set."""
 
-        PortSeed.n = self.options.port_seed
-
         check_json_precision()
 
         self.options.cachedir = os.path.abspath(self.options.cachedir)